test/BUILD.gn

```diff
@@ -1028,6 +1027,10 @@ test("brave_browser_tests") {
     deps += [ "//brave/browser/playlist/test:browser_tests" ]
   }
 
+  if (enable_brave_referrals) {
+    deps += [ "//brave/components/brave_referrals/browser" ]
```

Comment:
the fact that dep is conditional, but source files are not is weird. I've looked into this and found that we should also move `brave/browser/brave_stats/brave_stats_updater_browsertest.cc` into this condition.

OR we should make `brave_stats_updater_browsertest.cc` work without referrals service (if possible).